### PR TITLE
feat!: remove blobs from data

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

This PR removes blobs from the data struct. The tx field thus contains both sdk txs and blobtxs. This is part of ADR 20 which no longer requires the separation of blobs from transactions.

Note that for execution and for the tx indexer, the PFBs are extracted from the BlobTxs and are executed/indexed

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use
  [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by evan-forbes — APPROVED** [review]
nice! does this mean that we keep everything as a blob tx, or we have some other interpretation of that data?

File: proto/tendermint/types/types.proto [hunk 1]
```diff
@@ -89,11 +89,10 @@ message Data {
...
   repeated bytes txs = 1;
 
+  reserved 2, 3, 4;
   // field number 2 is reserved for intermediate state roots
-  // field number 3 was previously used for evidence
-
-  // Blobs are the data blobs included in the original data square.
-  repeated Blob blobs       = 4 [(gogoproto.nullable) = false];
+  // field number 3 is reserved for evidence
+  // field number 4 is reserved for blobs
```
Comment (rootulp) [inline proto/tendermint/types/types.proto:95]: 🆒 use of `reserved`. Should we consider adding

```proto
reserved "intermediate_state_roots", "evidence", "blobs";
```

Ref: https://protobuf.dev/programming-guides/proto3/#fieldreserved

**Review by rootulp — APPROVED** [review]
Yay for minimizing the diff w/ CometBFT

**Comment by cmwaters** [comment]
> nice! does this mean that we keep everything as a blob tx, or we have some other interpretation of that data?

Yes everything remains as a blobtx.

**Pull request merged**